WordPress/gutenberg
# Allow Opt-out from HTTP V1 Middleware in API Fetch

## What?
This PR adds a function to completely opt out from the HttpV1 Middleware in the API Fetch.

## Why?
Since the HTTP V1 spec is very old and most clients today use HTTP 1.1 which is backwards compatible. So we can allow generally used HTTP methods such as PUT, PATCH, DELETE.
Reference Issue - https://github.com/WordPress/gutenberg/issues/67655

## How?
I have added a function to remove the middleware. This helps in opt out as per user requirements.

## Testing Instructions
1. Create a new custom block.
2. Add a fetch request using API Fetch and set the method between PUT, PATCH or DELETE.
3. Before making the request ensure that the function `removeHttpV1Middleware` is called.
4. Perform the request to determine the final result.
